File: cases/taint/python/nested-access-path-positive/nested_access_path.py
def dfb_source():  # DFB-SOURCE: nested-access-path-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: nested-access-path-sink
    pass


class Inner:
    def __init__(self):
        self.value = "clean"
        self.other = "clean"


class Middle:
    def __init__(self):
        self.inner = Inner()


class Outer:
    def __init__(self):
        self.middle = Middle()


def run():
    outer = Outer()
    outer.middle.inner.value = dfb_source()  # DFB-WITNESS: nested-access-path-store
    dfb_sink(outer.middle.inner.value)

File: cases/taint/ruby/nested-access-path-positive/nested_access_path.rb
class Inner
  attr_accessor :value, :other

  def initialize
    @value = "clean"
    @other = "clean"
  end
end

class Middle
  attr_accessor :inner

  def initialize
    @inner = Inner.new
  end
end

class Outer
  attr_accessor :middle

  def initialize
    @middle = Middle.new
  end
end

def dfb_source # DFB-SOURCE: nested-access-path-input
  "tainted"
end

def dfb_sink(value) # DFB-SINK: nested-access-path-sink
end

def run
  outer = Outer.new
  outer.middle.inner.value = dfb_source # DFB-WITNESS: nested-access-path-store
  dfb_sink(outer.middle.inner.value)
end
